native/effektio/src/platform/android.rs

```diff
@@ -12,19 +23,137 @@ pub async fn new_client_config(base_path: String, home: String) -> Result<Client
     Ok(builder)
 }
 
-pub fn init_logging(filter: Option<String>) -> Result<()> {
-    std::env::set_var("RUST_BACKTRACE", "1");
+static mut FILE_LOGGER: Option<Arc<fern::ImplDispatch>> = None;
 
+pub fn init_logging(app_name: String, log_dir: String, filter: Option<String>) -> Result<()> {
+    std::env::set_var("RUST_BACKTRACE", "1");
     log_panics::init();
 
+    let log_level = match filter {
+        Some(ref filter) => FilterBuilder::new().parse(&filter).build(),
+        None => FilterBuilder::new().build(),
+    };
+
     let mut log_config = Config::default()
-        .with_min_level(Level::Trace)
-        .with_tag("effektio-sdk");
+        .with_max_level(LevelFilter::Trace)
+        .with_tag(app_name.as_str());
     if let Some(filter) = filter {
-        log_config = log_config.with_filter(FilterBuilder::new().parse(&filter).build())
+        log_config = log_config.with_filter(FilterBuilder::new().parse(&filter).build());
     }
+    let console_logger = LoggerWrapper::new(log_config).cloned_boxed_logger();
 
-    android_logger::init_once(log_config);
+    let mut path = PathBuf::from(log_dir.as_str());
+    path.push("app_");
+
+    let (level, dispatch) = fern::Dispatch::new()
+        .format(|out, message, record| {
+            out.finish(format_args!(
+                "{}[{}][{}] {}",
+                chrono::Local::now().format("[%Y-%m-%d][%H:%M:%S]"),
+                record.target(),
+                record.level(),
+                message
+            ))
+        })
+        // Only log messages Info and above
+        .level(log_level.filter())
+        // Output to console
+        .chain(console_logger)
+        // Output to file
+        .chain(fern::Manual::new(path, "%Y-%m-%d_%H-%M-%S%.f.log"))
+        .into_dispatch_with_arc();
+
+    if level == log::LevelFilter::Off {
+        log::set_boxed_logger(Box::new(native::NopLogger)).unwrap();
+    } else {
+        log::set_boxed_logger(Box::new(dispatch.clone())).unwrap();
+    }
+    log::set_max_level(level);
+
+    unsafe {
+        FILE_LOGGER = Some(dispatch);
+    }
```

Comment:
Please use a global mutex and/or lazy, rather than `unsafe` here, please.

Reply:
Yeah, I fixed that by `lazy_static`.
Please reference https://github.com/acterglobal/a3/pull/488/commits/91869ec5f90d3912b76ec5034ff404afb4b42127